internal/expr/query.go

```diff
@@ -119,6 +121,11 @@ func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) (scanArgs []an
 		mapVal  reflect.Value
 	}
 	var mapSetInfos = []mapSetInfo{}
+	type scanInfo struct {
+		pointerScanVal reflect.Value
+		fieldVal       reflect.Value
+	}
+	var scanInfos = []scanInfo{}
```

Comment:
These should be initialized as nil:
```
var mapSetInfos []mapSetInfo
var scanInfos []scanInfo
```
Also, the nomenclature here is a bit uneven. Both of these are doing exactly the same thing: having a proxy value in the scan phase, so it can be set into the final value on success. So maybe we could try to unify:
```
type scanProxy struct {
        scan reflect.Value
        real reflect.Value
        key  reflect.Value
}
var scanProxies []scanProxy
```
Then we can probably unify the logic for both. If not a map, we just don't set the key.

Reply:
Logic unified.